### DISCUSSION TITLE
Ip is banned but not working

### DISCUSSION THREAD

**Discussion by pierrotori** [fail2ban/fail2ban#3584] [General]
Hi
i installed the last version version-1.0.2-r0 armv71 on docker and for domoticz filter
the system detect the ip and ban the ip
`Status for the jail: domoticz
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     2
|  - File list:        /remotelogs/domoticz/domoticz.log
- Actions
   |- Currently banned: 1
   |- Total banned:     1
  - Banned IP list:   92.184.99.175`

and on the host 
`sudo iptables-legacy -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
f2b-domoticz  tcp  --  anywhere             anywhere             multiport dports https,8443

Chain f2b-domoticz (1 references)
target     prot opt source               destination
REJECT     all  --  92-184-99-175.mobile.fr.orangecustomers.net  anywhere             reject-with icmp-port-unreachable
RETURN     all  --  anywhere             anywhere`

but the problem , i continue to connect on the domoticz service

is the problem is in the docker-compose file, or the alternative on the iptables instead of iptables-legacy ?
i needed your help 
Regards

**Comment by sebres** [comment]
See the answer for a 2nd question in [wiki : FAQ](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works)...

That can be affected by many reasons: from wrong port to whitelisting rules allowing already established connections.
But since I see DOCKER, I guess you have something like this - #2698 (e. g. you have to supply chain to action as `DOCKER-USER` instead of `INPUT`) or this - #2376.

Also check whether you see errors in fail2ban.log by ban or jail start.

**Comment by pierrotori** [comment]
my config
pi@raspberrypi:~ $ sudo iptables-legacy -S |grep f2b
-N f2b-domoticz
-A f2b-domoticz -j RETURN

pi@raspberrypi:~ $ sudo iptables -S |grep DOCKER-USER
 Warning: iptables-legacy tables present, use iptables-legacy to see them
-N DOCKER-USER
-A FORWARD -j DOCKER-USER
-A DOCKER-USER -j RETURN

if i ban a ip with te parameter chain= DOCKER-USER

i have this error in the fail2ban.log
2023-09-28 19:38:14,984 76F9B45C NOTIC [domoticz] Ban 192.168.0.3
 2023-09-28 19:38:15,048 76F9B45C ERROR 763346c0 -- exec: { iptables -w -C f2b-domoticz -j RETURN >/dev/null 2>&1; } || { iptables -w -N f2b-domoticz || true; iptables -w -A f2b-domoticz -j RETURN; }
for proto in $(echo 'tcp' | sed 's/,/ /g'); do
{ iptables -w -C DOCKER-USER -p $proto -m multiport --dports https,8443 -j f2b-domoticz >/dev/null 2>&1; } || { iptables -w -I DOCKER-USER -p $proto -m multiport --dports https,8443 -j f2b-domoticz; }
done
 2023-09-28 19:38:15,050 76F9B45C ERROR 763346c0 -- stderr: 'iptables: No chain/target/match by that name.'
 2023-09-28 19:38:15,051 76F9B45C ERROR 763346c0 -- returned 1
 2023-09-28 19:38:15,051 76F9B45C ERROR Failed to execute ban jail 'domoticz' action 'iptables-multiport' info 'ActionInfo({'ip': '192.168.0.3', 'family': 'inet4', 'fid': <function Actions.ActionInfo.<lambda> at 0x764c02f8>, 'raw-ticket': <function Actions.ActionInfo.<lambda> at 0x764c0658>})': Error starting action Jail('domoticz')/iptables-multiport: 'Script error'
 2023-09-28 19:48:15,358 75C5ED84 NOTIC [domoticz] Unban 192.168.0.3

**Comment by pierrotori** [comment]
i find the problem, i force the iptables-legacy and executre the command
sudo update-alternatives --set iptables /usr/sbin/iptables-legacy
now it's on legacy and the FORWARD rule work fine

pi@raspberrypi:~ $ sudo iptables-legacy -S |grep DOCKER-USER
-N DOCKER-USER
-A FORWARD -j DOCKER-USER
-A DOCKER-USER -p tcp -m multiport --dports 443,8443 -j f2b-domoticz
-A DOCKER-USER -j RETURN

pi@raspberrypi:~ $ sudo iptables-legacy -S |grep f2b
-N f2b-domoticz
-A DOCKER-USER -p tcp -m multiport --dports 443,8443 -j f2b-domoticz
-A f2b-domoticz -s 92.184.106.27/32 -j REJECT --reject-with icmp-port-unreachable
-A f2b-domoticz -s 192.168.0.3/32 -j REJECT --reject-with icmp-port-unreachable
-A f2b-domoticz -j RETURN

all the rules are a same iptables-legacy, thanks
